src/libYARP_sig/src/Image.cpp

```diff
@@ -669,33 +707,30 @@ void Image::resize(int imgWidth, int imgHeight) {
     if (imgQuantum!=((ImageStorage*)implementation)->extern_type_quantum) {
         change = true;
     }
-    if (size!=imgPixelSize) {
-        imgPixelSize = size;
-        change=true;
-    }
+
     if (imgWidth!=width()||imgHeight!=height()) {
         change = true;
     }
 
     if (change) {
         ((ImageStorage*)implementation)->resize(imgWidth,imgHeight,
                                                 imgPixelCode,
-                                                imgPixelSize,
                                                 imgQuantum,
                                                 topIsLow);
         synchronize();
         //printf("CHANGE! %ld\n", (long int)(this));
     }
 }
 
+void Image::setPixelSize(int imgPixelSize) {
+    if(imgPixelSize == getPixelSize())
+        return;
 
-void Image::setPixelCode(int imgPixelCode) {
-    this->imgPixelCode = imgPixelCode;
+    return setPixelCode(-imgPixelSize);
```

Comment:
??

Reply:
@Nicogene Don't worry, this is pure magic... 🤣 
@aerydna Please add some comments to explain this